pkg/config/parser_test.go

```diff
@@ -30,131 +30,10 @@ provider "aws" {
-  connection {
-    dsn =  "postgres://postgres:pass@localhost:5432/postgres"
-  }
-  provider "test" {
-    version = "v0.0.0"
-  }
-}
-
-provider "test" {
-  configuration {
-	account "dev" {
-		role_arn ="12312312"
-	}
-	account "ron" {}
-  }
-  resources = ["slow_resource"]
-}`
-
-const testAliasProviderConfig = `cloudquery {
-  connection {
-    dsn =  "postgres://postgres:pass@localhost:5432/postgres"
-  }
-  provider "test" {
-    source = "cloudquery"
-    version = "v0.0.0"
-  }
-}
-
-provider "aws" {
-  configuration {
-	account "dev" {
-		role_arn ="12312312"
-	}
-	account "ron" {}
-  }
-  resources = ["slow_resource"]
-}
-
-provider "aws" {
-  alias = "another-aws"
-  configuration {
-	account "dev" {
-		role_arn ="12312312"
-	}
-	account "ron" {}
-  }
-  resources = ["slow_resource"]
-}`
+const expectedDuplicateProviderError = "fixtures/duplicate_provider_name.hcl:21,1-15: Provider Alias Required; Provider with name aws already exists, use alias in provider configuration block."
 
-const testMultipleProviderConfig = `cloudquery {
-  connection {
-    dsn =  "postgres://postgres:pass@localhost:5432/postgres"
-  }
-  provider "test" {
-    source = "cloudquery"
-    version = "v0.0.0"
-  }
-}
-
-provider "aws" {
-  configuration {
-	account "dev" {
-		role_arn ="12312312"
-	}
-	account "ron" {}
-  }
-  resources = ["slow_resource"]
-}
-
-provider "aws" {
-  configuration {
-	account "dev" {
-		role_arn ="12312312"
-	}
-	account "ron" {}
-  }
-  resources = ["slow_resource"]
-}
-`
-const expectedDuplicateProviderError = "test.hcl:21,1-15: Provider Alias Required; Provider with name aws already exists, use alias in provider configuration block."
-
-const testDuplicateAliasProviderConfig = `cloudquery {
-  connection {
-    dsn =  "postgres://postgres:pass@localhost:5432/postgres"
-  }
-  provider "test" {
-    source = "cloudquery"
-    version = "v0.0.0"
-  }
-}
-
-provider "aws" {
-  alias = "same-aws"
-  configuration {
-	account "dev" {
-		role_arn ="12312312"
-	}
-	account "ron" {}
-  }
-  resources = ["slow_resource"]
-}
-
-provider "aws" {
-  alias = "same-aws"
-  configuration {
-	account "dev" {
-		role_arn ="12312312"
-	}
-	account "ron" {}
-  }
-  resources = ["slow_resource"]
-}
-`
-const expectedDuplicateAliasProviderError = "test.hcl:23,3-21: Duplicate Alias; Provider with alias same-aws for provider aws already exists, give it a different alias."
-
-const testBadVersion = `cloudquery {
-  connection {
-    dsn =  "postgres://postgres:pass@localhost:5432/postgres"
-  }
-  provider "test" {
-    source = "cloudquery"
-    version = "0.0.0"
-  }
-}`
+const expectedDuplicateAliasProviderError = "fixtures/duplicate_provider_alias.hcl:23,3-21: Duplicate Alias; Provider with alias same-aws for provider aws already exists, give it a different alias."
+const expectedDuplicateAliasProviderErrorYaml = "<empty>:0,0-0: Duplicate Alias; Provider with alias same-aws for provider aws-2 already exists, give it a different alias."
```

Comment:
This is something that's missing from YAML. We don't parse it block by block, so we don't get exact line numbers